x-pack/legacy/plugins/maps/public/layers/vector_layer.js

```diff
@@ -585,6 +596,34 @@ export class VectorLayer extends AbstractLayer {
     });
   }
 
+  _setMbTextProperties(mbMap) {
+    const sourceId = this.getId();
+    const textLayerId = this._getMbTextLayerId();
+    const textLayer = mbMap.getLayer(textLayerId);
+
+    if (!textLayer) {
+      mbMap.addLayer({
+        id: textLayerId,
+        type: 'symbol',
+        source: sourceId,
+      });
+    }
+
+    const filterExpr = getPointFilterExpression(this._hasJoins());
+    if (filterExpr !== mbMap.getFilter(textLayerId)) {
+      mbMap.setFilter(textLayerId, filterExpr);
+    }
+
+    this._style.setMBPropertiesForText({
+      isLayerVisible: this.isVisible(),
```

Comment:
this flag seems like a micro-optimization where we're hacking the visibility property, and the approach is different from all the other props. 

If we want to delay visibility of the labels based on `this._labelStyleProperty.isComplete()`, then create a null style-rule or something when the label-config is incomplete

Reply:
This is used to hide the mapbox source when users hide a layer. Without this then the text will not be hidden when a user hides a layer.